src/users/users.controller.spec.ts

```diff
@@ -2,21 +2,90 @@ import { Test, TestingModule } from '@nestjs/testing';
 import { UsersController } from './users.controller';
 import { UsersService } from './users.service';
 import { TestModule } from 'src/shared/testkits';
+import { UsersAdminsController } from './users.admin.controller';
+import { DBModule, User } from 'src/shared/schema';
+import {
+  ApplicationStatus,
+  RegistrationMethod,
+  UserRole,
+  UserStatus,
+} from 'src/shared/interfaces';
 
-describe('UsersController', () => {
+describe('UsersAdminController', () => {
   let controller: UsersController;
+  let adminController: UsersAdminsController;
+  let usersService: UsersService;
 
   beforeEach(async () => {
     const module: TestingModule = await Test.createTestingModule({
-      imports: [TestModule],
-      controllers: [UsersController],
+      imports: [TestModule, DBModule],
+      controllers: [UsersController, UsersAdminsController],
       providers: [UsersService],
     }).compile();
 
     controller = module.get<UsersController>(UsersController);
+    adminController = module.get<UsersAdminsController>(UsersAdminsController);
+    usersService = module.get<UsersService>(UsersService);
   });
 
   it('should be defined', () => {
-    expect(controller).toBeDefined();
+    expect(adminController).toBeDefined();
+  });
+
+  describe('findByUsername', () => {
+    let requestMock = { email: 'test@example.com' };
+    it('should return true', async () => {
+      const userMock = createUserMock({
+        email: 'test@example.com',
```

Comment:
Great work, we need to perform both positive and negative test scenarios.

1. to test when the username is found
2. when the username is not found

Reply:
yes. I intend to do that , but wanted to get fix for the unit tests out there